cmd/triggerrun/cmd/root.go

```diff
@@ -88,24 +93,57 @@ func trigger(triggerFile, httpPath string) error {
 		return fmt.Errorf("fail to get clients: %w", err)
 	}
 
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return fmt.Errorf("Failed to build config from the flags: %w", err)
+	}
+
 	logger, _ := zap.NewProduction()
 	sugerLogger := logger.Sugar()
 	eventID := template.UID()
+	sinkClients, err := sink.ConfigureClients(config)
+	if err != nil {
+		log.Fatalf("Failed to get the sink client: %v", err)
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to get the dynamic client: %v", err)
+	}
+
+	dynamicCS := dynamicClientset.New(tekton.WithClient(dynamicClient))
+	r := sink.Sink{
+		KubeClientSet:          kubeClient,
+		HTTPClient:             http.DefaultClient,
+		Auth:                   sink.DefaultAuthOverride{},
+		DiscoveryClient:        sinkClients.DiscoveryClient,
+		DynamicClient:          dynamicCS,
+		Logger:                 sugerLogger,
+		EventListenerNamespace: "default",
+	}
+
 	for _, tri := range triggers {
 		eventLog := sugerLogger.With(zap.String(triggersv1.EventIDLabelKey, eventID))
 		resources, err := processTriggerSpec(kubeClient, triggerClient, tri,
-			request, body, eventID, eventLog)
+			request, body, eventID, eventLog, r)
 		if err != nil {
 			return fmt.Errorf("fail to create resources: %w", err)
 		}
 
-		for _, resource := range resources {
-			s, err := yaml.Marshal(resource)
+		if action == "show" {
+			for _, resource := range resources {
+				s, err := yaml.Marshal(resource)
+				if err != nil {
+					return fmt.Errorf("fail to print out the resource: %w", err)
+				}
+				fmt.Println("-----------------------------------")
+				fmt.Printf("%s", s)
+			}
+		} else if action == "create" {
```

Comment:
Will there be any other actions?
If yes can we make use of switch case instead of if/else WDYT?

Reply:
It only has two actions, which is "Create" and "Show" for now, but it will have more actions in in the future.